src/segm8.cpp

```diff
@@ -298,16 +301,25 @@ void SegM8::display(String string, uint8_t position, uint8_t width, uint8_t flag
 
     uint8_t trueLength = 0; // used chars in the input buffer
     uint8_t trueWidth = 0; // length of result
-    for (i = 0; true; i++) {
-        if (string.length() <= i) // end of string
-            break;
-        else if (trueWidth == width) // calculation end
-            break;
-        else if (string[i] == '.') {
-            trueWidth--;
+
+    uint8_t prev = 0;
```

Comment:
Don't repeat yourself. Can duplicated code be moved into a separate function?

Reply:
Good idea. Done. Look down.